src/NServiceBus.Core/Recoverability/SecondLevelRetries/DelayedRetryExecutor.cs

```diff
@@ -29,11 +29,11 @@ public Task Retry(IncomingMessage message, TimeSpan delay, ContextBag context)
             outgoingMessage.Headers[Headers.RetriesTimestamp] = DateTimeExtensions.ToWireFormattedString(DateTime.UtcNow);
 
             UnicastAddressTag messageDestination;
-            DeliveryConstraint[] deliveryConstraints = null;
+            List<DeliveryConstraint> deliveryConstraints = null;
             if (timeoutManagerAddress == null)
             {
                 // transport supports native deferred messages, directly send to input queue with delay constraint:
-                deliveryConstraints = new DeliveryConstraint[]
+                deliveryConstraints = new List<DeliveryConstraint>
```

Comment:
since the collection initializer doesn't set the capacity property, should we do that here? (e.g. `new List<DeliveryConstraint>(1){...}`)